python/pyarrow/_compute.pyx

```diff
@@ -2020,17 +2025,22 @@ cdef class Expression(_Weakrefable):
         return (<Expression> Expression._scalar(expr))
 
     @staticmethod
-    cdef Expression _call(str function_name, list arguments,
-                          shared_ptr[CFunctionOptions] options=(
-                              <shared_ptr[CFunctionOptions]> nullptr)):
+    def _call(str function_name, list arguments, FunctionOptions options=None):
         cdef:
             vector[CExpression] c_arguments
+            shared_ptr[CFunctionOptions] c_options=(
+                <shared_ptr[CFunctionOptions]> nullptr)
```

Comment:
You don't need to explicitly initialize this, do you?

Reply:
Ah, yes, passing an unitialized `c_options` to CMakeCallExpression seems to work as well